packages/lisk-p2p/src/peer.ts

```diff
@@ -13,26 +13,42 @@
  *
  */
 
+import { EventEmitter } from 'events';
 import { RPCResponseError } from './errors';
+
 import {
 	P2PMessagePacket,
-	P2PNodeStatus,
+	P2PNodeInfo,
 	P2PRequestPacket,
 	P2PResponsePacket,
+	ProtocolInboundMessage,
+	ProtocolInboundRPCRequest,
 } from './p2p_types';
 
-import { create, SCClientSocket } from 'socketcluster-client';
+import socketClusterClient, { SCClientSocket } from 'socketcluster-client';
 import { SCServerSocket } from 'socketcluster-server';
 import { processPeerListFromResponse } from './response_handler_sanitization';
 
+// Local emitted events.
+export const EVENT_UPDATED_PEER_INFO = 'updatedPeerInfo';
+export const EVENT_INVALID_REQUEST_RECEIVED = 'invalidRequestReceived';
+export const EVENT_REQUEST_RECEIVED = 'requestReceived';
+export const EVENT_INVALID_MESSAGE_RECEIVED = 'invalidMessageReceived';
+export const EVENT_MESSAGE_RECEIVED = 'requestReceived';
+
+// Remote event or RPC names sent to or received from peers.
+export const REMOTE_EVENT_RPC_REQUEST = 'rpc-request';
+export const REMOTE_EVENT_SEND_NODE_INFO = 'updateMyself';
+export const REMOTE_RPC_GET_ALL_PEERS_LIST = 'list';
+
+
 export interface PeerInfo {
 	readonly ipAddress: string;
 	readonly wsPort: number;
-	readonly nodeStatus?: P2PNodeStatus; // TODO DELEEETETETE
 	readonly clock?: Date;
 	readonly height: number;
-	readonly os: string;
-	readonly version: string;
+	readonly os?: string;
+	readonly version?: string;
```

Comment:
That's a tricky one.

The reason why version is optional is because the list of seedPeers which is fetched from the config file has this format:

`readonly seedPeers: ReadonlyArray<PeerInfo>;`

We cannot know what the version number of the seed peer is before we connect to it.
I think it's OK to expect the `version` in the sanitization logic when checking inbound PeerInfo from a remote peer, but it shouldn't be compulsory because the `PeerInfo` object may be created directly in our own code or from a config file.

Not sure if it's better to add yet another `Peer...` type? It's convenient that `PeerInfo` can be used for both getting information about peers and also for creating peers.

Reply:
one idea to have `BasePeerInfo` which is extended to `SeedPeerIndo` and `PeerInfo` where the difference is version?

Reply:
@shuse2 Sounds good but I think that `SeedPeerInfo` might end up having the same properties as `BasePeerInfo` so maybe just `BasePeerInfo` and `PeerInfo`?

Or it could be `PeerInfo` (for constructing) and `DetailedPeerInfo` (when getting information out of the node with `getNetworkStatus()`)?

Reply:
> BasePeerInfo and PeerInfo

sounds good =)